rero_ils/modules/persons/api.py

```diff
@@ -64,16 +65,24 @@ class Person(IlsRecord):
     @classmethod
     def get_record_by_mef_pid(cls, pid):
         """Get record using MEF REST API."""
-        rec = cls.get_record_by_pid(pid)
-        if rec:
-            return rec
-        # No data found: request on MEF URL
-        data = cls._get_mef_record(pid)
-        # Register MEF person and index it
-        metadata = data.get('metadata')
-        if '$schema' in metadata:
-            del metadata['$schema']
-        rec = cls.create(metadata, dbcommit=True, reindex=True)
+        try:
+            with db.session.begin_nested():
+                rec = cls.get_record_by_pid(pid)
+                if rec:
+                    return rec
+                # No data found: request on MEF URL
+                data = cls._get_mef_record(pid)
+                # Register MEF person and index it
+                metadata = data.get('metadata')
+                if '$schema' in metadata:
+                    del metadata['$schema']
+                rec = cls.create(metadata)
+        except:
+            current_app.logger.error('ERROR get MEF record: {pid}'.format(
+                pid=pid
+            ))
+            return None
+        rec.reindex()
```

Comment:
Is db.session.commit() missing or it is not required?

Reply:
The `with` doing the magic stuff like `db.session.commit()`or `db.session.rollback()`.